tests/test_megaphone.py

```diff
@@ -0,0 +1,32 @@
+import mock
+import pytest
+
+from kinto_megaphone import megaphone
+
+
+@mock.patch('kinto_megaphone.megaphone.requests')
+def test_megaphone_send_version_does_put(requests):
+    m = megaphone.Megaphone('http://example.com/', 'mytoken')
+    m.send_version('bid', 'chanid', 'myversion')
+    assert requests.put.call_count == 1
+    requests.put.assert_called_with('http://example.com/v1/broadcasts/bid/chanid',
+                                    auth=megaphone.BearerAuth('mytoken'),
+                                    data='myversion')
+
+
+@mock.patch('kinto_megaphone.megaphone.requests')
+def test_megaphone_send_put_failure_raises(requests):
+    m = megaphone.Megaphone('http://example.com/', 'mytoken')
+    resp = requests.put.return_value = mock.Mock()
+    e = resp.raise_for_status.side_effect = AssertionError('hi')
+    with pytest.raises(AssertionError) as excinfo:
+        m.send_version('bid', 'chanid', 'myversion')
+    assert excinfo.value == e
```

Comment:
Does this rollback changes?

Reply:
I don't understand the question.

Reply:
If a failure occurs while submitting data to Megaphone, will the transaction be rollbacked or will the data be stored in DB and failure ignored?

Reply:
Writing a test for this is a bit tricky because the default storage backend is the memory one, which doesn't support rollbacks as far as I can tell.